core/src/poh.rs

```diff
@@ -32,6 +32,10 @@ impl Poh {
     }
 
     pub fn hash(&mut self, max_num_hashes: u64) -> bool {
+        if self.remaining_hashes == 1 {
```

Comment:
Does this really help?

Reply:
Not sure if it's helping. But it aligns well with ```record()```.

Reply:
Hmm, but removing this new block doesn't change the behavior of `hash()` in any way so if it's not clearly helping...